pyccel/commands/pyccel_make.py

```diff
@@ -10,159 +10,195 @@
 
 import argparse
 import glob
-import os
 import sys
 from pathlib import Path
 
-def pyccel_make_command() -> None:
+from .argparse_helpers import add_version_flag, add_compiler_selection, add_accelerator_selection
+from .argparse_helpers import add_common_settings, check_file_type
+
+PYCCEL_MAKE_DESCR = 'Translate and compile multiple Python files in a project.'
+
+class GlobAction(argparse.Action):
     """
-    Pyccel console command.
+    Class describing a glob argument.
+
+    Class describing a glob argument. The action saves the files matching
+    the glob to the specified destination. The type check is used on the
+    located files.
+
+    Parameters
+    ----------
+    option_strings : obj
+        See argparse.Action.
+    dest : str
+        The name of the attribute where the files are saved.
+    nargs : int, optional
+        The number of arguments that can be passed to this flag.
+    type : Function, optional
+        A function which can be used to check the type of the files.
+    **kwargs : dict
+        See argparse.Action.
+    """
+    def __init__(self, option_strings, dest, nargs=None, type=None, **kwargs): #pylint: disable=redefined-builtin
+        self._type_check = type
+        super().__init__(option_strings, dest, nargs=None, **kwargs)
 
-    The command line interface allowing pyccel-make to be called.
+    def __call__(self, parser, namespace, values, option_string=None):
+        # Get files
+        files = [Path(f) for f in glob.glob(values, recursive=True)]
+
+        # Check types
+        try:
+            for f in files:
+                self._type_check(f)
+        except argparse.ArgumentTypeError as err:
+            raise argparse.ArgumentError(self, message=err) from err
```

Comment:
Would it be possible to put the try/except block inside the loop, collect all the errors, and print them all at the end? That could become really verbose, though..

Reply:
I'm not sure how to concatenate errors cleanly. I think there is more to an error than just the message.

> That could become really verbose, though..

I agree. I also think people would check all their files if they see a problem with one